backend/prompt_studio/prompt_studio_output_manager/output_manager_helper.py

```diff
@@ -144,3 +145,39 @@ def fetch_default_llm_profile(tool: CustomTool) -> ProfileManager:
             return ProfileManager.get_default_llm_profile(tool=tool)
         except DefaultProfileError:
             raise DefaultProfileError("Default ProfileManager does not exist.")
+
+    @staticmethod
+    def fetch_default_response(
+        tool_studio_prompts: list[ToolStudioPrompt], document_manager_id: str
+    ) -> dict[str, Any]:
+        # Initialize the result dictionary
+        result: dict[str, Any] = {}
+        # Iterate over ToolStudioPrompt records
+        for tool_prompt in tool_studio_prompts:
+            if tool_prompt.prompt_type == PSOMKeys.NOTES:
+                continue
+            prompt_id = str(tool_prompt.prompt_id)
+            profile_manager_id = tool_prompt.profile_manager_id
+
+            # If profile_manager is not set, skip this record
+            if not profile_manager_id:
+                result[tool_prompt.prompt_key] = ""
+                continue
+
+            try:
+                queryset = PromptStudioOutputManager.objects.filter(
+                    prompt_id=prompt_id,
+                    profile_manager=profile_manager_id,
+                    is_single_pass_extract=False,
+                    document_manager_id=document_manager_id,
+                )
+
+                if not queryset.exists():
+                    result[tool_prompt.prompt_key] = ""
+                    continue
+
+                for output in queryset:
+                    result[tool_prompt.prompt_key] = output.output
```

Comment:
@jagadeeswaran-zipstack Can you add your inputs to the [discussion](https://github.com/Zipstack/unstract/pull/513#discussion_r1689070317)?